www/public/repl/repl.js

```diff
@@ -165,11 +165,37 @@ async function processInputQueue() {
 // Callbacks to JS from Rust
 // ----------------------------------------------------------------------------
 
+var ROC_PANIC_INFO = null;
+
+function send_panic_msg_to_js(rocstr_ptr, tag_id) {
+    const { memory } = repl.app.exports;
+    const memoryBuffer = new Uint8Array(memory.buffer);
+
+    const rocStrBytes = new Uint8Array(memory.buffer, rocstr_ptr, 12);
+
+    const strPtr = rocStrBytes[0] | (rocStrBytes[1] << 8) | (rocStrBytes[2] << 16) | (rocStrBytes[3] << 24);
+    const strLen = rocStrBytes[4] | (rocStrBytes[5] << 8) | (rocStrBytes[6] << 16) | (rocStrBytes[7] << 24);
```

Comment:
based on some searches there is no better way to achieve this?!

Reply:
You can do something like this (please test for bugs!)
```js
    const rocStrWords = new Uint32Array(memory.buffer, rocstr_ptr, 3);
    const [ptr, len, cap] = rocStrWords;
```

Reply:
The main idea being to use Uint32Array rather than Uint8Array
These classes are "views" onto an ArrayBuffer, but interpret the buffer as either u32 or u8 respectively.
The ArrayBuffer itself can't really do much.

Docs here
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint32Array/Uint32Array

The second argument to the constructor is called byteOffset in the docs. Third argument is length, which I think is the length of the final Uint32Array but test it!

Reply:
Also `memoryBuffer` appears to be unused?
You might have un-pushed changes because I see you marked one of my comments as resolved but I don't see the changes.

Reply:
huh, I tried the "transmute" from bytes before and then it did not work. maybe I used the wrong length. Anyway that does in fact work now